modules/xphoto/src/inpainting.cpp

```diff
@@ -66,6 +77,820 @@ namespace cv
+                            xblock_offset = 1;
+                        }
+                        else if (quader_counter == 2)
+                        {
+                            yblock_offset = 1;
+                            xblock_offset = 0;
+                        }
+                        else if (quader_counter == 3)
+                        {
+                            yblock_offset = 1;
+                            xblock_offset = 1;
+                        }
+
+                        set_later.emplace_back(xblock_counter_quadernary + xblock_offset, yblock_counter_quadernary + yblock_offset);
+                    }
+
+                }
+            }
+
+        }
+    }
+
+
+    static void
+    icvDetermineProcessingOrder(const cv::Mat& _sampled_img, const cv::Mat& _sampling_mask, const int quality, const std::string& channel, cv::Mat& reconstructed_img)
+    {
+        fsr_parameters fsr_params(quality);
+        int block_size = fsr_params.block_size;
+        int block_size_max = fsr_params.block_size;
+        int block_size_min = fsr_params.block_size_min;
+        double conc_weighting = fsr_params.conc_weighting;
+        int fft_size = fsr_params.fft_size;
+        double rho = fsr_params.rhos[0];
+        cv::Mat sampled_img, sampling_mask;
+        _sampled_img.convertTo(sampled_img, CV_64F);
+        reconstructed_img = sampled_img.clone();
+
+        int height = _sampling_mask.rows;
+        int width = _sampling_mask.cols;
+        uchar* maskData = (uchar*)_sampling_mask.data;
+        for (int y = 0; y < height; ++y)
+        {
+            for (int x = 0; x < width; ++x)
+            {
+                if (maskData[y*_sampling_mask.step1() + x] > 0)
+                {
+                    maskData[y*_sampling_mask.step1() + x] = 1;
+                }
+                else
+                {
+                    maskData[y*_sampling_mask.step1() + x] = 0;
+                }
+            }
+        }
+        _sampling_mask.convertTo(sampling_mask, CV_64F);
+
+        double threshold_stddev_LUT[3];
+        if (channel == "Y")
+        {
+            std::copy(fsr_params.threshold_stddev_Y, fsr_params.threshold_stddev_Y + 3, threshold_stddev_LUT);
+        }
+        else if (channel == "Cx")
+        {
+            std::copy(fsr_params.threshold_stddev_Cx, fsr_params.threshold_stddev_Cx + 3, threshold_stddev_LUT);
+        }
+        else
+        {
+            CV_Error(cv::Error::StsBadArg, "channel type unsupported!");
+        }
+
+
+        double threshold_stddev = threshold_stddev_LUT[0];
+
+        std::vector< std::tuple< int, int > > set_later;
+        int img_height = sampled_img.rows;
+        int img_width = sampled_img.cols;
+
+        // inital scan of distorted blocks
+        std::vector< std::tuple< int, int > > set_todo;
+        int blocks_column = cvCeil(static_cast<double>(img_height) / block_size);
+        int blocks_line = cvCeil(static_cast<double>(img_width) / block_size);
```

Comment:
We can avoid floating-point computations here.
Use `int divUp(int a, unsigned int b)` from `core/utility.hpp`